cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go

```diff
@@ -235,6 +235,21 @@ func (t Template) ControllersPolicy() *infrav1.PolicyDocument {
 			})
 		}
 	}
+	if t.Spec.S3Buckets.Enable {
+		statement = append(statement, infrav1.StatementEntry{
+			Effect: infrav1.EffectAllow,
+			Resource: infrav1.Resources{
+				"arn:*:s3:::*",
+			},
+			Action: infrav1.Actions{
+				"s3:CreateBucket",
```

Comment:
Looking at #2662, I now wonder if maybe S3 IAM should also be part of separate policy/role, similar to what we're going to have for EKS support.

Reply:
I think we need to balance between having a huge policy which exceeds the AWS limit and having too many policies attached to a user/role/group. By default you can attach up to 10 managed policies to an IAM role for example.

Reply:
Right now the S3-related policy is rather small. The EKS policy OTOH is quite big. Let's decide to separate the S3 policy if/when it becomes significantly bigger.

Reply:
Also, let's see if the approach in #2662 is acceptable to the maintainers.